history/taskana-simplehistory-provider/src/test/java/acceptance/events/task/CreateHistoryEventOnTaskCancelClaimAccTest.java

```diff
@@ -48,8 +53,45 @@ void should_CreateCancelClaimedHistoryEvent_When_TaskIsCancelClaimed() throws Ex
 
     assertThat(events).hasSize(1);
 
-    String eventType = events.get(0).getEventType();
+    TaskHistoryEvent event = events.get(0);
+    String eventType = event.getEventType();
 
     assertThat(eventType).isEqualTo(TaskHistoryEventType.CLAIM_CANCELLED.getName());
+
+    event = historyService.getTaskHistoryEvent(event.getId());
+
+    assertThat(event.getDetails()).isNotNull();
+
+    JSONArray changes = new JSONObject(event.getDetails()).getJSONArray("changes");
+
+    assertThat(changes.length()).isPositive();
+    for (int i = 0; i < changes.length(); i++) {
+      JSONObject change = changes.getJSONObject(i);
+      switch (change.get("fieldName").toString()) {
+        case "received":
+          assertThat(change.get("oldValue")).isEqualTo("");
+          assertThat(change.get("newValue")).isEqualTo(task.getReceived().toString());
+          break;
+        case "claimed":
+          assertThat(change.get("oldValue")).isEqualTo(oldClaimed.toString());
+          assertThat(change.get("newValue")).isEqualTo("");
+          break;
+        case "modified":
+          assertThat(change.get("oldValue")).isEqualTo(oldModified.toString());
+          assertThat(change.get("newValue")).isEqualTo(task.getModified().toString());
+          break;
+        case "state":
+          assertThat(change.get("oldValue")).isEqualTo("CLAIMED");
+          assertThat(change.get("newValue")).isEqualTo("READY");
+          break;
+        case "owner":
+          assertThat(change.get("oldValue")).isEqualTo("user-b-1");
+          assertThat(change.get("newValue")).isEqualTo("");
+          break;
+        default:
+          assertThat(change.get("oldValue")).isEqualTo(change.get("newValue"));
+          break;
+      }
+    }
```

Comment:
I don't like this way of validating the changes array, because this is only a partial validation. Furthermore this style is not compatible with our testing guidelines since this does not use the assertj fluent api.
E.g. the changes array only contains e.g. "state" =>  this test will pass even tho that change should be part of the array.

One way of implementing the fluent api is to create the exact content of the JSONArray and use its equals method.